### PR TITLE
create apis for create and list tasks

### PR DESCRIPTION
**Pull request by Walaa-Volidis**

### CODE REVIEW

File: src/app/api/createTask/route.ts [hunk 1]
```diff
@@ -0,0 +1,31 @@
...
+  description: z.string(),
+  category: z.string(),
+  date: z.string().datetime(),
+  userId: z.string(),
+});
+export type Task = z.infer<typeof ZTaskSchema>;
+export async function POST(request: Request) {
+  try {
+    const formData = await request.formData();
+    const task: Task = {
```
Comment (nmaswood) [inline src/app/api/createTask/route.ts:16]:
```suggestion
    const task = ZTaskSchema.parse({
          title: formData.get("title"),
      description: formData.get("description"),
      category: formData.get("category"),
      date: formData.get("date"),
      userId: formData.get("userId"),
      
      })
```

File: src/app/api/createTask/route.ts [hunk 2]
```diff
@@ -0,0 +1,31 @@
...
+      title: formData.get("title") as string,
+      description: formData.get("description") as string,
+      category: formData.get("category") as string,
+      date: formData.get("date") as string,
+      userId: formData.get("userId") as string,
+    };
+    ZTaskSchema.parse(task);
```
Comment (nmaswood) [inline src/app/api/createTask/route.ts:23]: ✂️

File: src/app/api/createTask/route.ts [hunk 3]
```diff
@@ -0,0 +1,31 @@
+import prisma from "@/lib/prisma";
+import { z } from "zod";
+
+const ZTaskSchema = z.object({
+  id: z.number().optional(),
```
Comment (nmaswood) [inline src/app/api/createTask/route.ts:5]: ✂️ You can remove the id field here right?

File: src/app/api/listTasks/route.ts [hunk 4]
```diff
@@ -0,0 +1,24 @@
...
+export async function GET() {
+  const response = await prisma.task.findMany();
+
+  const formattedResponse = response.map((task) => ({
+    ...task,
+    date: task.date.toISOString(),
+  }));
+
+  const tasksArray = ZTaskSchema.array();
+  const validatedResponse = tasksArray.parse(formattedResponse);
```
Comment (nmaswood) [inline src/app/api/listTasks/route.ts:22]:
```suggestion
  const validatedResponse = ZTaskSchema.array().parse(formattedResponse);
```

**Review by nmaswood — APPROVED** [review]
LGTM

File: src/app/api/createTask/route.ts (hunk 1 above)
Comment (Walaa-Volidis) [reply src/app/api/createTask/route.ts:16]: done

**Pull request merged**